pkg/controllers/management/node/cleanup.go

```diff
@@ -393,11 +397,30 @@ func (m *Lifecycle) createCleanupJob(userContext *config.UserContext, cluster *v
 	return userContext.K8sClient.BatchV1().Jobs("default").Create(context.TODO(), &job, metav1.CreateOptions{})
 }
 
+func (m *Lifecycle) userNodeRemoveCleanup(obj *v3.Node) (runtime.Object, error) {
+	newObj := obj.DeepCopy()
+	newObj.SetFinalizers(removeFinalizerWithPrefix(newObj.GetFinalizers(), userNodeRemoveFinalizerPrefix))
+
+	if newObj.DeletionTimestamp == nil {
+		annos := newObj.GetAnnotations()
+		if annos == nil {
+			annos = make(map[string]string)
+		} else {
+			annos = removeAnnotationWithPrefix(annos, userNodeRemoveAnnotationPrefix)
+			delete(annos, userNodeRemoveCleanupAnnotationOld)
+		}
+
+		annos[userNodeRemoveCleanupAnnotation] = "true"
+		newObj.SetAnnotations(annos)
+	}
+	return m.nodeClient.Update(newObj)
+}
+
 func removeFinalizerWithPrefix(finalizers []string, prefix string) []string {
 	var nf []string
 	for _, finalizer := range finalizers {
 		if strings.HasPrefix(finalizer, prefix) {
-			logrus.Debugf("a finalizer with prefix %s will be removed", prefix)
+			logrus.Debugf("finalizer with prefix %s will be removed", prefix)
```

Comment:
nit: put brackets around resource names `finalizer with prefix [%s] will be removed`